### PR TITLE
add ros2 param

### PR DESCRIPTION
**Pull request by dirk-thomas**
* `list` outputs the names of all parameters, either for all nodes or a node passed as an argument
* `get` outputs the value of a specific parameter, the type information can be suppressed
* `set` sets a parameter, the type is being determined by guessing the primitive types, no array types atm
* `delete` removed a specific parameter (at least it should but it seems that the service doesn't work for that use case).

### CODE REVIEW

File: ros2param/ros2param/verb/list.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                for node_name in [n for n in node_names if n not in futures]:
+                    # call as soon as ready
+                    client = clients[node_name]
+                    if client.service_is_ready():
+                        request = ListParameters.Request()
+                        print(node_name, request)
+                        future = client.call_async(request)
+                        futures[node_name] = future
+                    else:
+                        print(node_name, 'not ready')
```
Comment (sloretz) [inline ros2param/ros2param/verb/list.py]: Looks like there are a few debug print statements. Use the node's logger for debug messages instead?

Comment (dirk-thomas) [reply ros2param/ros2param/verb/list.py]: I removed the debug output in 55a5948af40f1cd8989599d1c0c4bd9205ceb05e.

**Comment by mikaelarguedas** [comment]
Tested this along https://github.com/ros2/rclcpp/pull/489 and it works great :+1: 
(I haven't tested with many requests queued up, only one a a time)

I pushed a few nitpicks, feel free to revert them if they're not appropriate.

A few questions that come to mind:

- Could we add the option to provide the type explicitly ?
  - e.g. if I want to pass to a string parameter a value that can be converted to another type ("0", "True", "42.42"...)
- Is there something in the API preventing to sex the array types ? or was it just to keep the scope reasonable?

**Comment by dirk-thomas** [comment]
> * Could we add the option to provide the type explicitly ?
> * Is there something in the API preventing to set the array types ? or was it just to keep the scope reasonable?

Additional feature can be added anytime. You could add it to this PR right now or keep it for a future PR. I just wanted to get a first version of this verb in to be able to use / test the parameters more easily.

**Comment by mikaelarguedas** [comment]
> Additional feature can be added anytime. You could add it to this PR right now or keep it for a future PR. I just wanted to get a first version of this verb in to be able to use / test the parameters more easily.

Follow up is fine by me :+1: it's great to have a first version of this verb out!
I just wanted to make sure that there was not a technical blocker for implementing all parameter types in the other verbs, I didn't mean to appear requesting additional features in this PR :).

**Pull request merged**